### ISSUE TITLE
Experiment with querying mlmd data

### ISSUE THREAD

**Issue by antisaling** [opendatahub-io/model-registry-bf4-kf#7]
HuggingFace makes tags available for searching on their interface. Whilst we might deserialize a complex structure (say, with JSON, as per internal discussions) it might be best to experiment with nested artifacts to accomplish this functionality with simpler processing (e.g. handling only the initial query to pass it to the proper DB entry).

**Comment by dhirajsb** [comment]
We can just use a string with tags separated by `,`, and then use a query like `tags LIKE '%foo%'`.

**Comment by antisaling** [comment]
I have experimented with lists as comma-separated strings, and holding references to nested artifacts with their keys. The approach I came up with is to use suffixes on the property keys to indicate what the value being held actually is:
- `.key` indicates the property is a key to another artifact
- `.key_list` indicates it is a (comma-separated) list of such keys
- `.list` indicates it is a comma-separated string of scalars

This takes away some flexibility as we need to know the suffix of a given property in advance, either by checking all available properties so we can match a non-suffixed entry to its actual key, or by trying out different combinations until one succeeds. It also takes away some flexibility as we can't import values that have such suffixes in them as it'd cause problems parsing the data incorrectly. I haven't experimented further but one possible workaround would be to prefix the data itself, then parse after accessing it, but this comes with its own problems.

Experiments were done at: https://github.com/isinyaaa/hf-mlmd-mapping-poc
